spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -35,6 +35,9 @@ import org.apache.comet.expressions.{CometCast, Compatible}
 class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
   import testImplicits._
 
+  /** Create a data generator using a fixed seed so that tests are reproducible */
+  private val gen = new DataGenerator(new Random(42))
```

Comment:
To make sure I understand what you are suggesting, I think you are saying to add this:


```
object DataGenerator {
  val DEFAULT = new DataGenerator(new Random(42))
}
```

And then reference that from the test suite?

```
private val gen = DataGenerator.DEFAULT
```

I think this could lead to some non-deterministic behavior. For example, running `CometCastSuite` on its own from my IDE would start with the freshly created generator, but running the same suite after after test suites have run would result in different inputs.

I think it would be better for each test to create a new instance of the generator with a fixed seed?

Reply:
I suppose we could define it as a `def` rather than a `val` to force a new creation each time. WDYT?

Reply:
Thanks for the review @advancedxy. I have implemented your suggestion.